cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/ETLConfig.java

```diff
@@ -41,20 +41,23 @@
 // though not marked nullable, fields can be null since these objects are created through gson deserialization
 @SuppressWarnings("ConstantConditions")
 public class ETLConfig extends Config implements UpgradeableConfig {
-  private final String description;
-  private final Set<ETLStage> stages;
-  private final Set<Connection> connections;
-  private final Resources resources;
-  private final Resources driverResources;
-  private final Resources clientResources;
-  private final Boolean stageLoggingEnabled;
-  private final Boolean processTimingEnabled;
-  private final Integer numOfRecordsPreview;
-  private final Map<String, String> properties;
+  protected final String description;
```

Comment:
why do these need to be made protected?

Reply:
So that we dont have to use default getter and setter which are setting default value when null. But for upgrade config, we want to serialize and deserialize without considering/changing values to default. If its null in original, it should be null in result.

Reply:
And I needed to use them in ETLBatchConfig class so.